File: example/java/jimmer-sql-graphql/src/main/java/org/babyfish/jimmer/sql/example/graphql/dal/AuthorRepository.java
package org.babyfish.jimmer.sql.example.graphql.dal;

import org.babyfish.jimmer.spring.repository.JRepository;
import org.babyfish.jimmer.sql.ast.Predicate;
import org.babyfish.jimmer.sql.example.graphql.entities.Author;
import org.babyfish.jimmer.sql.example.graphql.entities.AuthorTable;
import org.springframework.lang.Nullable;
import org.springframework.util.StringUtils;

import java.util.List;

public interface AuthorRepository extends JRepository<Author, Long> {

    AuthorTable table = AuthorTable.$;

    default List<Author> findByName(@Nullable String name) {

        return sql()
                .createQuery(table)
                .whereIf(
                        StringUtils.hasText(name),
                        Predicate.or(
                                table.firstName().ilike(name),
                                table.lastName().ilike(name)
                        )
                )
                .orderBy(table.firstName(), table.lastName())
                .select(table)
                .execute();
    }
}

File: example/kotlin/jimmer-sql-graphql-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/graphql/dal/AuthorRepository.kt
package org.babyfish.jimmer.example.kt.graphql.dal

import org.babyfish.jimmer.example.kt.graphql.entities.*
import org.babyfish.jimmer.spring.repository.KRepository
import org.babyfish.jimmer.sql.kt.ast.expression.ilike
import org.babyfish.jimmer.sql.kt.ast.expression.or

interface AuthorRepository : KRepository<Author, Long> {

    fun find(name: String?): List<Author> =
        sql.createQuery(Author::class) {
            name?.let {
                where(
                    or(
                        table.firstName ilike it,
                        table.lastName ilike it
                    )
                )
            }
            select(table)
        }.execute()
}